compiler/rustc_attr_parsing/src/context.rs

```diff
@@ -260,7 +260,8 @@ impl<'sess> AttributeParser<'sess> {
                 // }
                 ast::AttrKind::Normal(n) => {
                     let parser = MetaItemParser::from_attr(n, self.dcx());
-                    let (path, args) = parser.deconstruct();
```

Comment:
I'm not sure this is an improvement

Reply:
since it's a reasonably common operation I think

Reply:
It doesn't have many uses, which is why the second commit got rid of it. But no matter, I've reinstated the old code here, we can discuss `deconstruct` more in the follow-up.